core-rust/mesh-api-server/src/mesh_api/handlers/construction_parse.rs

```diff
@@ -0,0 +1,197 @@
+    assert_matching_network(&request.network_identifier, &state.network)?;
+
+    let transaction_bytes = hex::decode(&request.transaction).map_err(|_| {
+        ResponseError::from(ApiError::InvalidTransaction)
+            .with_details(format!("Invalid transaction hex: {}", &request.transaction))
+    })?;
+    let (instructions, signers) = if request.signed {
+        let transaction =
+            NotarizedTransactionV1::from_raw(&RawNotarizedTransaction::from_vec(transaction_bytes))
+                .map_err(|_| {
+                    ResponseError::from(ApiError::InvalidTransaction)
+                        .with_details(format!("Invalid transaction: {}", &request.transaction))
+                })?;
+        let validated = transaction
+            .prepare_and_validate(&TransactionValidator::new_with_latest_config(
+                &state.network,
+            ))
+            .map_err(|e| {
+                ResponseError::from(ApiError::InvalidTransaction)
+                    .with_details(format!("Invalid transaction: error = {:?}", e))
+            })?;
+
+        let instructions = transaction.signed_intent.intent.instructions.0;
+        let signers = validated.signer_keys;
+        (instructions, signers)
+    } else {
+        let prepared_intent = PreparedIntentV1::prepare(
+            &RawTransactionIntent::from_vec(transaction_bytes),
+            &PreparationSettings::latest(),
+        )
+        .unwrap();
+        #[allow(deprecated)]
+        let instructions = prepared_intent.instructions.inner.0;
+        let signers = index_set_new();
+        (instructions, signers)
+    };
+
+    let database = state.state_manager.database.snapshot();
+    let operations = parse_instructions(
+        &instructions,
+        &MappingContext::new(&state.network),
+        database.deref(),
+    )?;
+
+    // See https://docs.cdp.coinbase.com/mesh/docs/models#constructionparseresponse for field
+    // definitions
+    Ok(Json(models::ConstructionParseResponse {
+        operations,
+        signers: None,
+        account_identifier_signers: Some(
+            signers
+                .into_iter()
+                .map(|x| AccountIdentifier {
+                    address: state.public_key_to_address_string(x),
+                    sub_account: None,
+                    metadata: None,
+                })
+                .collect(),
+        ),
+        metadata: None,
+    }))
+}
+
+pub fn parse_instructions(
+    instructions: &[InstructionV1],
+    mapping_context: &MappingContext,
+    database: &StateManagerDatabase<impl ReadableRocks>,
+) -> Result<Vec<Operation>, ResponseError> {
+    let mut operations = Vec::new();
+    let mut next_index = 0;
+    while next_index < instructions.len() {
+        let mut instruction = &instructions[next_index];
+        next_index = next_index + 1;
+        match instruction {
+            InstructionV1::CallMethod(CallMethod {
```

Comment:
We should add a filter on this branch to `if global_address.is_account_address()`

Reply:
Done